borg/repository.py

```diff
@@ -83,6 +85,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is not None:
             self.rollback()
+            if hasattr(exc_val, 'errno'):
+                if exc_val.errno in [errno.EIO, errno.ENXIO]:
+                    return
         self.close()
```

Comment:
If what you mean by _I don't see the difference to your earlier approach?_ is the use of return instead of re-raise, note that `__exit__()` methods should not reraise the passed-in exception (Reference: Python Documentation) since the exception will be processed normally upon exit from this method.
Furthermore, return was used to prevent further exceptions to be raised due to the call of `self.close()` in the case of repository/cache related OSError.
It's true that the exceptions that "travel by" this method could've been raised pretty much anywhere. Do you think `errno.EIO` and `errno.ENXIO` could have been caused by other than repository/cache related OSError?